{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "Tratamento_dados_surdez.ipynb",
      "provenance": [],
      "collapsed_sections": [
        "DP-p4zRT1nr3",
        "KsMplAKi2Yq5"
      ]
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "code",
      "metadata": {
        "id": "u1wBLrE4y-qi"
      },
      "source": [
        "import pandas as pd\n",
        "import numpy as np\n",
        "import gzip"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gAifZdvm9JMI",
        "outputId": "fd778cb7-772c-4ea8-d3f2-1a48647dde4c"
      },
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "JVcEjI1lz4go"
      },
      "source": [
        "### Clinvar variants treatment \n",
        "\n",
        "- Pathogenic, likely pathogenic and conflicting variants selection"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 490
        },
        "id": "bsOBR838zWcm",
        "outputId": "c3fb978c-cae4-4f8e-f352-77a990d60e49"
      },
      "source": [
        "with gzip.open('/content/drive/MyDrive/GRAR/Paper_Surdez/clinvar.vcf.gz','rt') as file:\n",
        "  count=0\n",
        "  for line in file:\n",
        "    if line.startswith('#'):\n",
        "      count+=1\n",
        "    else:\n",
        "      break\n",
        "print(count)\n",
        "\n",
        "df_clinvar = pd.read_table('/content/drive/MyDrive/GRAR/Paper_Surdez/clinvar.vcf.gz', header=None,skiprows=count)\n",
        "df_clinvar_total = df_clinvar.copy()\n",
        "df_clinvar"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "28\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2822: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
            "  if self.run_code(code, result):\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>2</th>\n",
              "      <th>3</th>\n",
              "      <th>4</th>\n",
              "      <th>5</th>\n",
              "      <th>6</th>\n",
              "      <th>7</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>925952</td>\n",
              "      <td>1019397</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>ALLELEID=1003021;CLNDISDB=MedGen:CN517202;CLND...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>930139</td>\n",
              "      <td>1125147</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>ALLELEID=1110865;CLNDISDB=MedGen:CN517202;CLND...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>930165</td>\n",
              "      <td>1164676</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>ALLELEID=1153701;CLNDISDB=MedGen:CN517202;CLND...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>930187</td>\n",
              "      <td>1144630</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>ALLELEID=1131738;CLNDISDB=MedGen:CN517202;CLND...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1</td>\n",
              "      <td>930188</td>\n",
              "      <td>846933</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>ALLELEID=824438;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083268</th>\n",
              "      <td>MT</td>\n",
              "      <td>16013</td>\n",
              "      <td>690274</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>ALLELEID=677666;CLNDISDB=MONDO:MONDO:0010789,M...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083269</th>\n",
              "      <td>MT</td>\n",
              "      <td>16015</td>\n",
              "      <td>690276</td>\n",
              "      <td>T</td>\n",
              "      <td>TATTCTCTGTTCTTTC</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>ALLELEID=677667;CLNDISDB=MONDO:MONDO:0010789,M...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083270</th>\n",
              "      <td>MT</td>\n",
              "      <td>16017</td>\n",
              "      <td>690275</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>ALLELEID=677668;CLNDISDB=MONDO:MONDO:0010789,M...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083271</th>\n",
              "      <td>MT</td>\n",
              "      <td>16021</td>\n",
              "      <td>690277</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>ALLELEID=677669;CLNDISDB=MONDO:MONDO:0010789,M...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083272</th>\n",
              "      <td>NW_009646201.1</td>\n",
              "      <td>83614</td>\n",
              "      <td>17735</td>\n",
              "      <td>TC</td>\n",
              "      <td>T</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>ALLELEID=32774;CLNDISDB=MedGen:C0000778,OMIM:6...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>1083273 rows Ã— 8 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "                      0  ...                                                  7\n",
              "0                     1  ...  ALLELEID=1003021;CLNDISDB=MedGen:CN517202;CLND...\n",
              "1                     1  ...  ALLELEID=1110865;CLNDISDB=MedGen:CN517202;CLND...\n",
              "2                     1  ...  ALLELEID=1153701;CLNDISDB=MedGen:CN517202;CLND...\n",
              "3                     1  ...  ALLELEID=1131738;CLNDISDB=MedGen:CN517202;CLND...\n",
              "4                     1  ...  ALLELEID=824438;CLNDISDB=MedGen:CN517202;CLNDN...\n",
              "...                 ...  ...                                                ...\n",
              "1083268              MT  ...  ALLELEID=677666;CLNDISDB=MONDO:MONDO:0010789,M...\n",
              "1083269              MT  ...  ALLELEID=677667;CLNDISDB=MONDO:MONDO:0010789,M...\n",
              "1083270              MT  ...  ALLELEID=677668;CLNDISDB=MONDO:MONDO:0010789,M...\n",
              "1083271              MT  ...  ALLELEID=677669;CLNDISDB=MONDO:MONDO:0010789,M...\n",
              "1083272  NW_009646201.1  ...  ALLELEID=32774;CLNDISDB=MedGen:C0000778,OMIM:6...\n",
              "\n",
              "[1083273 rows x 8 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 4
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 771
        },
        "id": "Pkx6kqZ505zA",
        "outputId": "61b538b0-65e0-47d8-b74b-176b2f1498eb"
      },
      "source": [
        "df_clinvar = df_clinvar[df_clinvar[7].str.contains('CLNSIG=Pathogenic|CLNSIG=Likely_pathogenic|CLNSIG=Conflicting_interpretations_of_pathogenicity')]\n",
        "regions_regex = r'.*CLNVC=(.*?);.*MC=SO.*?\\|(.*?)[\\,|\\;].*'  \n",
        "df_clinvar[['var_type','mol_consequence']] = df_clinvar[7].str.extract(regions_regex,expand=True)\n",
        "df_clinvar = df_clinvar[[0,1,3,4,'var_type','mol_consequence']]\n",
        "df_clinvar.columns = ['chr','pos','ref','alt','var_type','mol_consequence']\n",
        "df_clinvar.head(20)\n"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/local/lib/python3.7/dist-packages/pandas/core/frame.py:3069: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  self[k1] = value[k2]\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>chr</th>\n",
              "      <th>pos</th>\n",
              "      <th>ref</th>\n",
              "      <th>alt</th>\n",
              "      <th>var_type</th>\n",
              "      <th>mol_consequence</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>322</th>\n",
              "      <td>1</td>\n",
              "      <td>943995</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>376</th>\n",
              "      <td>1</td>\n",
              "      <td>964512</td>\n",
              "      <td>C</td>\n",
              "      <td>A</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>missense_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>391</th>\n",
              "      <td>1</td>\n",
              "      <td>1013983</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>splice_acceptor_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>414</th>\n",
              "      <td>1</td>\n",
              "      <td>1014143</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>435</th>\n",
              "      <td>1</td>\n",
              "      <td>1014316</td>\n",
              "      <td>C</td>\n",
              "      <td>CG</td>\n",
              "      <td>Duplication</td>\n",
              "      <td>frameshift_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>444</th>\n",
              "      <td>1</td>\n",
              "      <td>1014359</td>\n",
              "      <td>G</td>\n",
              "      <td>T</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>478</th>\n",
              "      <td>1</td>\n",
              "      <td>1020239</td>\n",
              "      <td>G</td>\n",
              "      <td>C</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>missense_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>496</th>\n",
              "      <td>1</td>\n",
              "      <td>1022225</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>missense_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>504</th>\n",
              "      <td>1</td>\n",
              "      <td>1022313</td>\n",
              "      <td>A</td>\n",
              "      <td>T</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>missense_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>540</th>\n",
              "      <td>1</td>\n",
              "      <td>1040679</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>synonymous_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>541</th>\n",
              "      <td>1</td>\n",
              "      <td>1040694</td>\n",
              "      <td>G</td>\n",
              "      <td>GC</td>\n",
              "      <td>Duplication</td>\n",
              "      <td>frameshift_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>543</th>\n",
              "      <td>1</td>\n",
              "      <td>1040717</td>\n",
              "      <td>G</td>\n",
              "      <td>GGGGCC</td>\n",
              "      <td>Microsatellite</td>\n",
              "      <td>frameshift_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>555</th>\n",
              "      <td>1</td>\n",
              "      <td>1040819</td>\n",
              "      <td>G</td>\n",
              "      <td>GC</td>\n",
              "      <td>Duplication</td>\n",
              "      <td>frameshift_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>596</th>\n",
              "      <td>1</td>\n",
              "      <td>1041338</td>\n",
              "      <td>TCCTGCGCCGCG</td>\n",
              "      <td>T</td>\n",
              "      <td>Deletion</td>\n",
              "      <td>frameshift_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>600</th>\n",
              "      <td>1</td>\n",
              "      <td>1041354</td>\n",
              "      <td>CGCCCGCCAGGAGAATGTCTTCAAGAAGTTCGACG</td>\n",
              "      <td>C</td>\n",
              "      <td>Deletion</td>\n",
              "      <td>frameshift_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>614</th>\n",
              "      <td>1</td>\n",
              "      <td>1041559</td>\n",
              "      <td>G</td>\n",
              "      <td>GGCCC</td>\n",
              "      <td>Duplication</td>\n",
              "      <td>frameshift_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>618</th>\n",
              "      <td>1</td>\n",
              "      <td>1041582</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>632</th>\n",
              "      <td>1</td>\n",
              "      <td>1041678</td>\n",
              "      <td>CGCTCCGGCCAGTGCCAGGGTCGAGGTGAGCGGCTCCCCCGGGGGAGG</td>\n",
              "      <td>C</td>\n",
              "      <td>Deletion</td>\n",
              "      <td>splice_donor_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>663</th>\n",
              "      <td>1</td>\n",
              "      <td>1042053</td>\n",
              "      <td>C</td>\n",
              "      <td>G</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>672</th>\n",
              "      <td>1</td>\n",
              "      <td>1042136</td>\n",
              "      <td>T</td>\n",
              "      <td>TC</td>\n",
              "      <td>Duplication</td>\n",
              "      <td>frameshift_variant</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "    chr      pos  ...                   var_type          mol_consequence\n",
              "322   1   943995  ...  single_nucleotide_variant                 nonsense\n",
              "376   1   964512  ...  single_nucleotide_variant         missense_variant\n",
              "391   1  1013983  ...  single_nucleotide_variant  splice_acceptor_variant\n",
              "414   1  1014143  ...  single_nucleotide_variant                 nonsense\n",
              "435   1  1014316  ...                Duplication       frameshift_variant\n",
              "444   1  1014359  ...  single_nucleotide_variant                 nonsense\n",
              "478   1  1020239  ...  single_nucleotide_variant         missense_variant\n",
              "496   1  1022225  ...  single_nucleotide_variant         missense_variant\n",
              "504   1  1022313  ...  single_nucleotide_variant         missense_variant\n",
              "540   1  1040679  ...  single_nucleotide_variant       synonymous_variant\n",
              "541   1  1040694  ...                Duplication       frameshift_variant\n",
              "543   1  1040717  ...             Microsatellite       frameshift_variant\n",
              "555   1  1040819  ...                Duplication       frameshift_variant\n",
              "596   1  1041338  ...                   Deletion       frameshift_variant\n",
              "600   1  1041354  ...                   Deletion       frameshift_variant\n",
              "614   1  1041559  ...                Duplication       frameshift_variant\n",
              "618   1  1041582  ...  single_nucleotide_variant                 nonsense\n",
              "632   1  1041678  ...                   Deletion     splice_donor_variant\n",
              "663   1  1042053  ...  single_nucleotide_variant                 nonsense\n",
              "672   1  1042136  ...                Duplication       frameshift_variant\n",
              "\n",
              "[20 rows x 6 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 5
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 204
        },
        "id": "D6QYlJKSto7r",
        "outputId": "33a1dced-2fc2-49bd-8259-926e3a02ec17"
      },
      "source": [
        "df_clinvar['end'] = np.where( (df_clinvar['ref'].str.len() > df_clinvar['alt'].str.len() ) , df_clinvar['pos'].astype(int)+1, df_clinvar['pos'].astype(int)+ df_clinvar['alt'].str.len())\n",
        "df_clinvar = df_clinvar[['chr', 'pos', 'end', 'ref', 'alt','var_type','mol_consequence']]\n",
        "df_clinvar.columns = ['Chromosome','Start','End','Ref','Alt','var_type','mol_consequence']\n",
        "df_clinvar.head()"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "      <th>End</th>\n",
              "      <th>Ref</th>\n",
              "      <th>Alt</th>\n",
              "      <th>var_type</th>\n",
              "      <th>mol_consequence</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>322</th>\n",
              "      <td>1</td>\n",
              "      <td>943995</td>\n",
              "      <td>943996</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>376</th>\n",
              "      <td>1</td>\n",
              "      <td>964512</td>\n",
              "      <td>964513</td>\n",
              "      <td>C</td>\n",
              "      <td>A</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>missense_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>391</th>\n",
              "      <td>1</td>\n",
              "      <td>1013983</td>\n",
              "      <td>1013984</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>splice_acceptor_variant</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>414</th>\n",
              "      <td>1</td>\n",
              "      <td>1014143</td>\n",
              "      <td>1014144</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>435</th>\n",
              "      <td>1</td>\n",
              "      <td>1014316</td>\n",
              "      <td>1014318</td>\n",
              "      <td>C</td>\n",
              "      <td>CG</td>\n",
              "      <td>Duplication</td>\n",
              "      <td>frameshift_variant</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "    Chromosome    Start  ...                   var_type          mol_consequence\n",
              "322          1   943995  ...  single_nucleotide_variant                 nonsense\n",
              "376          1   964512  ...  single_nucleotide_variant         missense_variant\n",
              "391          1  1013983  ...  single_nucleotide_variant  splice_acceptor_variant\n",
              "414          1  1014143  ...  single_nucleotide_variant                 nonsense\n",
              "435          1  1014316  ...                Duplication       frameshift_variant\n",
              "\n",
              "[5 rows x 7 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 6
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "WpTjxzmU8it6",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 49
        },
        "outputId": "441144e2-de77-4854-ba04-0e2e4679bf8b"
      },
      "source": [
        "df_clinvar[df_clinvar.Chromosome == 'M']"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "      <th>End</th>\n",
              "      <th>Ref</th>\n",
              "      <th>Alt</th>\n",
              "      <th>var_type</th>\n",
              "      <th>mol_consequence</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "Empty DataFrame\n",
              "Columns: [Chromosome, Start, End, Ref, Alt, var_type, mol_consequence]\n",
              "Index: []"
            ]
          },
          "metadata": {},
          "execution_count": 7
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "oOsS3foX1aWH"
      },
      "source": [
        "### Opening deaf curated genes"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "i7-1Aask3g1m",
        "outputId": "8786b1d4-378d-45e1-c739-ff272d80f0dc"
      },
      "source": [
        "df_genes = pd.read_table('/content/drive/MyDrive/GRAR/Paper_Surdez/Genes_surdez.csv', sep=\";\", usecols=['Genes'])\n",
        "df_genes = df_genes.sort_values('Genes').drop_duplicates()\n",
        "print(f'Unique genes on the list -- {len(df_genes)}')"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Unique genes on the list -- 128\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "DP-p4zRT1nr3"
      },
      "source": [
        "### Opening MANE and RefSelect transcripts position\n",
        "- Comparing the transcripts with the genes list\n",
        "\n",
        "**results:**\n",
        "- 3 genes weren't found: MIR96, DFNX3, KCNJ10"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 142
        },
        "id": "MAootOxv07-W",
        "outputId": "15f5ee94-6244-425b-978e-6c741ad50b78"
      },
      "source": [
        "df_MANEbed = pd.read_table('/content/drive/MyDrive/GRAR/Paper_Surdez/canonical_transcripts_MANE7_sorted.bed',header=None)\n",
        "df_MANEbed.head(3)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>2</th>\n",
              "      <th>3</th>\n",
              "      <th>4</th>\n",
              "      <th>5</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr1</td>\n",
              "      <td>664</td>\n",
              "      <td>1237</td>\n",
              "      <td>PRAMEF22__NM_001100631.2_1__exon__1__chr1__f</td>\n",
              "      <td>0</td>\n",
              "      <td>+</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr1</td>\n",
              "      <td>65419</td>\n",
              "      <td>65433</td>\n",
              "      <td>OR4F5__NM_001005484.2__exon__1__chr1__f</td>\n",
              "      <td>0</td>\n",
              "      <td>+</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr1</td>\n",
              "      <td>65565</td>\n",
              "      <td>65573</td>\n",
              "      <td>OR4F5__NM_001005484.2__exon__2__chr1__f</td>\n",
              "      <td>0</td>\n",
              "      <td>+</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "      0      1      2                                             3  4  5\n",
              "0  chr1    664   1237  PRAMEF22__NM_001100631.2_1__exon__1__chr1__f  0  +\n",
              "1  chr1  65419  65433       OR4F5__NM_001005484.2__exon__1__chr1__f  0  +\n",
              "2  chr1  65565  65573       OR4F5__NM_001005484.2__exon__2__chr1__f  0  +"
            ]
          },
          "metadata": {},
          "execution_count": 9
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 142
        },
        "id": "492k8YJf5-YA",
        "outputId": "1e923417-2630-4f34-fdf9-8779facec50c"
      },
      "source": [
        "regions_regex = r'(.*)__(N.*)__(exon__[0-9]+)_'\n",
        "df_MANEbed[['gene', 'transcript', 'exon']] = df_MANEbed[3].str.extract(regions_regex, expand=True)\n",
        "df_MANEbed.drop(columns=[3,4,5], inplace = True)\n",
        "df_MANEbed.head(3)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>2</th>\n",
              "      <th>gene</th>\n",
              "      <th>transcript</th>\n",
              "      <th>exon</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr1</td>\n",
              "      <td>664</td>\n",
              "      <td>1237</td>\n",
              "      <td>PRAMEF22</td>\n",
              "      <td>NM_001100631.2_1</td>\n",
              "      <td>exon__1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr1</td>\n",
              "      <td>65419</td>\n",
              "      <td>65433</td>\n",
              "      <td>OR4F5</td>\n",
              "      <td>NM_001005484.2</td>\n",
              "      <td>exon__1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr1</td>\n",
              "      <td>65565</td>\n",
              "      <td>65573</td>\n",
              "      <td>OR4F5</td>\n",
              "      <td>NM_001005484.2</td>\n",
              "      <td>exon__2</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "      0      1      2      gene        transcript     exon\n",
              "0  chr1    664   1237  PRAMEF22  NM_001100631.2_1  exon__1\n",
              "1  chr1  65419  65433     OR4F5    NM_001005484.2  exon__1\n",
              "2  chr1  65565  65573     OR4F5    NM_001005484.2  exon__2"
            ]
          },
          "metadata": {},
          "execution_count": 10
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 142
        },
        "id": "y-0_FftmtMO4",
        "outputId": "f71b037f-5bc0-4866-ffbe-8038551dc013"
      },
      "source": [
        "df_merged = pd.merge(df_MANEbed, df_genes, left_on=['gene'], right_on=['Genes'], how='right', indicator= True)\n",
        "df_merged[df_merged['_merge'] == 'right_only']"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>2</th>\n",
              "      <th>gene</th>\n",
              "      <th>transcript</th>\n",
              "      <th>exon</th>\n",
              "      <th>Genes</th>\n",
              "      <th>_merge</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>483</th>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>DFNX3</td>\n",
              "      <td>right_only</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>857</th>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>KCNJ10</td>\n",
              "      <td>right_only</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>989</th>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>MIR96</td>\n",
              "      <td>right_only</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "       0   1   2 gene transcript exon    Genes      _merge\n",
              "483  NaN NaN NaN  NaN        NaN  NaN    DFNX3  right_only\n",
              "857  NaN NaN NaN  NaN        NaN  NaN  KCNJ10   right_only\n",
              "989  NaN NaN NaN  NaN        NaN  NaN    MIR96  right_only"
            ]
          },
          "metadata": {},
          "execution_count": 11
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "KsMplAKi2Yq5"
      },
      "source": [
        "### Opening RefSeq transcripts from Varstation Database\n",
        "- Comparing the transcripts with the genes list\n",
        "\n",
        "**results:**\n",
        "- 2 genes weren't found: DFNX3, KCNJ10 </br>\n",
        "- Assuming this dataframe as main\n",
        "- We are considering all possible transcripts positions\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 142
        },
        "id": "bnaY0x1evaWb",
        "outputId": "d9844ced-cb2e-4035-b194-f3b0b1290e33"
      },
      "source": [
        "df_transcript = pd.read_table('/content/drive/MyDrive/GRAR/Paper_Surdez/refSeq_transcript_regions_109.20210514.hg38.bed', header=None)\n",
        "df_transcript.head(3)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>2</th>\n",
              "      <th>3</th>\n",
              "      <th>4</th>\n",
              "      <th>5</th>\n",
              "      <th>6</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr1</td>\n",
              "      <td>11873</td>\n",
              "      <td>14409</td>\n",
              "      <td>DDX11L1</td>\n",
              "      <td>NR_046018.2</td>\n",
              "      <td>0</td>\n",
              "      <td>+</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr1</td>\n",
              "      <td>14361</td>\n",
              "      <td>29370</td>\n",
              "      <td>WASH7P</td>\n",
              "      <td>NR_024540.1</td>\n",
              "      <td>0</td>\n",
              "      <td>-</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr1</td>\n",
              "      <td>17368</td>\n",
              "      <td>17436</td>\n",
              "      <td>MIR6859-1</td>\n",
              "      <td>NR_106918.1</td>\n",
              "      <td>0</td>\n",
              "      <td>-</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "      0      1      2          3            4  5  6\n",
              "0  chr1  11873  14409    DDX11L1  NR_046018.2  0  +\n",
              "1  chr1  14361  29370     WASH7P  NR_024540.1  0  -\n",
              "2  chr1  17368  17436  MIR6859-1  NR_106918.1  0  -"
            ]
          },
          "metadata": {},
          "execution_count": 12
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 111
        },
        "id": "b-pOlXmNakBF",
        "outputId": "736f7cff-fa0e-46c9-806c-118ecd380bce"
      },
      "source": [
        "df_transcritos_merged = pd.merge(df_transcript, df_genes, left_on=[3], right_on=['Genes'], how='right', indicator= True)\n",
        "df_transcritos_merged[df_transcritos_merged['_merge'] == 'right_only']"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>2</th>\n",
              "      <th>3</th>\n",
              "      <th>4</th>\n",
              "      <th>5</th>\n",
              "      <th>6</th>\n",
              "      <th>Genes</th>\n",
              "      <th>_merge</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>85</th>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>DFNX3</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>DFNX3</td>\n",
              "      <td>right_only</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>188</th>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>KCNJ10</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>KCNJ10</td>\n",
              "      <td>right_only</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "       0   1   2        3    4   5    6    Genes      _merge\n",
              "85   NaN NaN NaN    DFNX3  NaN NaN  NaN    DFNX3  right_only\n",
              "188  NaN NaN NaN  KCNJ10   NaN NaN  NaN  KCNJ10   right_only"
            ]
          },
          "metadata": {},
          "execution_count": 13
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "nnaj1Cs2ggdp"
      },
      "source": [
        "df_transcritos_merged = df_transcritos_merged[df_transcritos_merged['_merge'] == 'both']\n",
        "df_transcritos_merged = df_transcritos_merged.iloc[:,:5]"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 204
        },
        "id": "2E1BR94ahefP",
        "outputId": "520501f8-003a-470b-a0b7-24e1272da8c8"
      },
      "source": [
        "df_transcritos_merged[0] = df_transcritos_merged[0].str.replace('chr','')\n",
        "df_transcritos_merged[[1,2]] = df_transcritos_merged[[1,2]].astype(int) #.str.replace('chr','')\n",
        "df_transcritos_merged.head()\n"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>2</th>\n",
              "      <th>3</th>\n",
              "      <th>4</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>16</td>\n",
              "      <td>15949751</td>\n",
              "      <td>16141281</td>\n",
              "      <td>ABCC1</td>\n",
              "      <td>NM_004996.4</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>17</td>\n",
              "      <td>81509970</td>\n",
              "      <td>81512799</td>\n",
              "      <td>ACTG1</td>\n",
              "      <td>NR_037688.3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>17</td>\n",
              "      <td>81510689</td>\n",
              "      <td>81512354</td>\n",
              "      <td>ACTG1</td>\n",
              "      <td>NM_001199954.3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>17</td>\n",
              "      <td>81510689</td>\n",
              "      <td>81512354</td>\n",
              "      <td>ACTG1</td>\n",
              "      <td>NM_001614.5</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>7</td>\n",
              "      <td>45574543</td>\n",
              "      <td>45713995</td>\n",
              "      <td>ADCY1</td>\n",
              "      <td>NM_021116.4</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "    0         1         2      3               4\n",
              "0  16  15949751  16141281  ABCC1     NM_004996.4\n",
              "1  17  81509970  81512799  ACTG1     NR_037688.3\n",
              "2  17  81510689  81512354  ACTG1  NM_001199954.3\n",
              "3  17  81510689  81512354  ACTG1     NM_001614.5\n",
              "4   7  45574543  45713995  ADCY1     NM_021116.4"
            ]
          },
          "metadata": {},
          "execution_count": 15
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 142
        },
        "id": "wByb86dXVWu5",
        "outputId": "19c13f3d-aafb-428d-b285-89f12ace54fa"
      },
      "source": [
        "df_transcritos_merged.columns = ['Chromosome','Start','End','Gene','Transcript']\n",
        "df_transcritos_merged.head(3)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "      <th>End</th>\n",
              "      <th>Gene</th>\n",
              "      <th>Transcript</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>16</td>\n",
              "      <td>15949751</td>\n",
              "      <td>16141281</td>\n",
              "      <td>ABCC1</td>\n",
              "      <td>NM_004996.4</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>17</td>\n",
              "      <td>81509970</td>\n",
              "      <td>81512799</td>\n",
              "      <td>ACTG1</td>\n",
              "      <td>NR_037688.3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>17</td>\n",
              "      <td>81510689</td>\n",
              "      <td>81512354</td>\n",
              "      <td>ACTG1</td>\n",
              "      <td>NM_001199954.3</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "  Chromosome     Start       End   Gene      Transcript\n",
              "0         16  15949751  16141281  ABCC1     NM_004996.4\n",
              "1         17  81509970  81512799  ACTG1     NR_037688.3\n",
              "2         17  81510689  81512354  ACTG1  NM_001199954.3"
            ]
          },
          "metadata": {},
          "execution_count": 16
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 142
        },
        "id": "jeXQREych48v",
        "outputId": "d418a632-6c61-4d93-e5d9-de8f88d40af9"
      },
      "source": [
        "df_transcritos_merged = df_transcritos_merged.groupby('Gene').agg({'Chromosome':'first','Start':'min', 'End':'max',}).reset_index()\n",
        "df_transcritos_merged.sort_values(['Chromosome','Start'], inplace=True)\n",
        "df_transcritos_merged.head(3)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Gene</th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "      <th>End</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>31</th>\n",
              "      <td>ESPN</td>\n",
              "      <td>1</td>\n",
              "      <td>6424955</td>\n",
              "      <td>6460146</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>40</th>\n",
              "      <td>GJB3</td>\n",
              "      <td>1</td>\n",
              "      <td>34784762</td>\n",
              "      <td>34785575</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>54</th>\n",
              "      <td>KCNQ4</td>\n",
              "      <td>1</td>\n",
              "      <td>40784093</td>\n",
              "      <td>40838523</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "     Gene Chromosome     Start       End\n",
              "31   ESPN          1   6424955   6460146\n",
              "40   GJB3          1  34784762  34785575\n",
              "54  KCNQ4          1  40784093  40838523"
            ]
          },
          "metadata": {},
          "execution_count": 17
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "E-aZFRnZ_gLu"
      },
      "source": [
        "#df_transcritos_merged.to_csv('/content/drive/MyDrive/GRAR/Paper_Surdez/transcripts_surdez.txt', index = None, header = False, sep = '\\t')"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 49
        },
        "id": "McYwFyeQQvQz",
        "outputId": "4d612c96-0283-470d-c94c-20fb76968ded"
      },
      "source": [
        "df_transcritos_merged[df_transcritos_merged.Chromosome == 'MT']"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Gene</th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "      <th>End</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "Empty DataFrame\n",
              "Columns: [Gene, Chromosome, Start, End]\n",
              "Index: []"
            ]
          },
          "metadata": {},
          "execution_count": 19
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "tGf5Ski73VCe"
      },
      "source": [
        "### Selecting Clinvar variants that fulfill the transcript positions\n",
        "\n",
        "**results:**\n",
        "- It's 5420 Pathogenic/Likely Pathogenic/Conflicting variants on deaf selected genes "
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "flPTogiXyAdL",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "d6510438-b617-426e-93a0-1846e2d949bc"
      },
      "source": [
        "!pip install pyranges\n",
        "import pyranges as pr"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: pyranges in /usr/local/lib/python3.7/dist-packages (0.0.112)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.7/dist-packages (from pyranges) (1.1.5)\n",
            "Requirement already satisfied: pyrle in /usr/local/lib/python3.7/dist-packages (from pyranges) (0.0.33)\n",
            "Requirement already satisfied: tabulate in /usr/local/lib/python3.7/dist-packages (from pyranges) (0.8.9)\n",
            "Requirement already satisfied: natsort in /usr/local/lib/python3.7/dist-packages (from pyranges) (5.5.0)\n",
            "Requirement already satisfied: cython in /usr/local/lib/python3.7/dist-packages (from pyranges) (0.29.24)\n",
            "Requirement already satisfied: sorted-nearest>=0.0.33 in /usr/local/lib/python3.7/dist-packages (from pyranges) (0.0.33)\n",
            "Requirement already satisfied: ncls>=0.0.63 in /usr/local/lib/python3.7/dist-packages (from pyranges) (0.0.63)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.7/dist-packages (from ncls>=0.0.63->pyranges) (1.19.5)\n",
            "Requirement already satisfied: pytz>=2017.2 in /usr/local/lib/python3.7/dist-packages (from pandas->pyranges) (2018.9)\n",
            "Requirement already satisfied: python-dateutil>=2.7.3 in /usr/local/lib/python3.7/dist-packages (from pandas->pyranges) (2.8.2)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/dist-packages (from python-dateutil>=2.7.3->pandas->pyranges) (1.15.0)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "WzMS74MN26W_"
      },
      "source": [
        "transcritos_pr = pr.PyRanges(df_transcritos_merged)\n",
        "clinvar_pr = pr.PyRanges(df_clinvar)"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "IeL6tKJHp_z0",
        "outputId": "89297e1d-5f0b-42d0-ca5b-3a224179b7a3"
      },
      "source": [
        "# Filtering and annotating clinvar variants \n",
        "annotate_transcript = clinvar_pr.join(transcritos_pr).drop(like=\"_b\")\n",
        "annotate_transcript"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "      <th>End</th>\n",
              "      <th>Ref</th>\n",
              "      <th>Alt</th>\n",
              "      <th>var_type</th>\n",
              "      <th>mol_consequence</th>\n",
              "      <th>Gene</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>6425219</td>\n",
              "      <td>6425220</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "      <td>ESPN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>6425247</td>\n",
              "      <td>6425248</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "      <td>ESPN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>6445936</td>\n",
              "      <td>6445937</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>splice_donor_variant</td>\n",
              "      <td>ESPN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>6448931</td>\n",
              "      <td>6448933</td>\n",
              "      <td>C</td>\n",
              "      <td>CG</td>\n",
              "      <td>Duplication</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>ESPN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1</td>\n",
              "      <td>6451602</td>\n",
              "      <td>6451603</td>\n",
              "      <td>G</td>\n",
              "      <td>C</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>splice_acceptor_variant</td>\n",
              "      <td>ESPN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5415</th>\n",
              "      <td>X</td>\n",
              "      <td>130147792</td>\n",
              "      <td>130147793</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>AIFM1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5416</th>\n",
              "      <td>X</td>\n",
              "      <td>130147804</td>\n",
              "      <td>130147805</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>AIFM1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5417</th>\n",
              "      <td>X</td>\n",
              "      <td>130149478</td>\n",
              "      <td>130149479</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>AIFM1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5418</th>\n",
              "      <td>X</td>\n",
              "      <td>130156576</td>\n",
              "      <td>130156577</td>\n",
              "      <td>G</td>\n",
              "      <td>C</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>AIFM1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5419</th>\n",
              "      <td>X</td>\n",
              "      <td>130165779</td>\n",
              "      <td>130165780</td>\n",
              "      <td>C</td>\n",
              "      <td>G</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>non-coding_transcript_variant</td>\n",
              "      <td>AIFM1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5420 rows Ã— 8 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "+--------------+-----------+-----------+------------+------------+-------+\n",
              "| Chromosome   | Start     | End       | Ref        | Alt        | +3    |\n",
              "| (category)   | (int32)   | (int32)   | (object)   | (object)   | ...   |\n",
              "|--------------+-----------+-----------+------------+------------+-------|\n",
              "| 1            | 6425219   | 6425220   | G          | A          | ...   |\n",
              "| 1            | 6425247   | 6425248   | C          | T          | ...   |\n",
              "| 1            | 6445936   | 6445937   | G          | A          | ...   |\n",
              "| 1            | 6448931   | 6448933   | C          | CG         | ...   |\n",
              "| ...          | ...       | ...       | ...        | ...        | ...   |\n",
              "| X            | 130147804 | 130147805 | G          | A          | ...   |\n",
              "| X            | 130149478 | 130149479 | C          | T          | ...   |\n",
              "| X            | 130156576 | 130156577 | G          | C          | ...   |\n",
              "| X            | 130165779 | 130165780 | C          | G          | ...   |\n",
              "+--------------+-----------+-----------+------------+------------+-------+\n",
              "Unstranded PyRanges object has 5,420 rows and 8 columns from 23 chromosomes.\n",
              "For printing, the PyRanges was sorted on Chromosome.\n",
              "3 hidden columns: var_type, mol_consequence, Gene"
            ]
          },
          "metadata": {},
          "execution_count": 22
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "yeIzxmOwyvZx"
      },
      "source": [
        "### Dealing with mitocondrial variants\n",
        "\n",
        "**results:**\n",
        "- From 120 variants only 22 were found at clinvar database \n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 204
        },
        "id": "-gZ1pcoVUti8",
        "outputId": "f86501fb-3ad6-4d23-8786-c104e283dbac"
      },
      "source": [
        "## Dataframe transcrito mitocondrial \n",
        "df_mitocond = df_transcript[df_transcript[0] == \"chrM\"].copy()\n",
        "df_mitocond.head()"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>2</th>\n",
              "      <th>3</th>\n",
              "      <th>4</th>\n",
              "      <th>5</th>\n",
              "      <th>6</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>77601</th>\n",
              "      <td>chrM</td>\n",
              "      <td>576</td>\n",
              "      <td>647</td>\n",
              "      <td>MT-TF</td>\n",
              "      <td>rna-TRNF</td>\n",
              "      <td>0</td>\n",
              "      <td>+</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>77602</th>\n",
              "      <td>chrM</td>\n",
              "      <td>647</td>\n",
              "      <td>1601</td>\n",
              "      <td>MT-RNR1</td>\n",
              "      <td>rna-RNR1</td>\n",
              "      <td>0</td>\n",
              "      <td>+</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>77603</th>\n",
              "      <td>chrM</td>\n",
              "      <td>1601</td>\n",
              "      <td>1670</td>\n",
              "      <td>MT-TV</td>\n",
              "      <td>rna-TRNV</td>\n",
              "      <td>0</td>\n",
              "      <td>+</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>77604</th>\n",
              "      <td>chrM</td>\n",
              "      <td>1670</td>\n",
              "      <td>3229</td>\n",
              "      <td>MT-RNR2</td>\n",
              "      <td>rna-RNR2</td>\n",
              "      <td>0</td>\n",
              "      <td>+</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>77605</th>\n",
              "      <td>chrM</td>\n",
              "      <td>3229</td>\n",
              "      <td>3304</td>\n",
              "      <td>MT-TL1</td>\n",
              "      <td>rna-TRNL1</td>\n",
              "      <td>0</td>\n",
              "      <td>+</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "          0     1     2        3          4  5  6\n",
              "77601  chrM   576   647    MT-TF   rna-TRNF  0  +\n",
              "77602  chrM   647  1601  MT-RNR1   rna-RNR1  0  +\n",
              "77603  chrM  1601  1670    MT-TV   rna-TRNV  0  +\n",
              "77604  chrM  1670  3229  MT-RNR2   rna-RNR2  0  +\n",
              "77605  chrM  3229  3304   MT-TL1  rna-TRNL1  0  +"
            ]
          },
          "metadata": {},
          "execution_count": 23
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "mQWzUQCFyuz6"
      },
      "source": [
        "df_mt = pd.read_table('/content/drive/MyDrive/GRAR/Paper_Surdez/MT_variants.txt',)\n",
        "df_mt['Position'] = df_mt['Position'].str.replace('1592C>6T','15926C>T') #correcting human issue\n",
        "regions_regex = (r'([0-9]+)([A-Z]+)>([A-Z]+)')\n",
        "\n",
        "df_mt[['Start','Ref','Alt']] = df_mt['Position'].str.extract(regions_regex,expand=True)\n"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "-LCAEP71KaKn"
      },
      "source": [
        "df_mt.loc[(df_mt.Position.str.contains(\"del|dup\")), ['Start']] = df_mt.Position.str.extract(r'([0-9]+)[del|dup].*')[0]\n",
        "df_mt.loc[(df_mt.Position.str.contains(\"del|dup\")), ['Ref']] = df_mt.Position.str.extract(r'.*[del|dup](.*)')[0]\n",
        "df_mt.loc[(df_mt.Position.str.contains(\"del\")), ['Alt']] = '.'\n",
        "df_mt.loc[(df_mt.Position.str.contains(\"dup\")), ['Alt']] = df_mt.Position.str.extract(r'.*dup(.*)')[0]*2\n",
        "df_mt= df_mt[['Gene','Position', 'Start','Ref','Alt']]\n",
        "\n"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "lPH7QQbdZSiY",
        "outputId": "5eb6a296-f21a-4aac-9a41-d4f56e407325"
      },
      "source": [
        "df_mt['End'] = np.where( (df_mt['Ref'].str.len() > df_mt['Alt'].str.len() ) , df_mt['Start'].astype(int)+1, df_mt['Start'].astype(int)+ df_mt['Alt'].str.len())\n",
        "df_mt['Chromosome'] = 'M'\n",
        "df_mt[['Start','End']] = df_mt[['Start','End']].astype(int)\n",
        "df_mt"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Gene</th>\n",
              "      <th>Position</th>\n",
              "      <th>Start</th>\n",
              "      <th>Ref</th>\n",
              "      <th>Alt</th>\n",
              "      <th>End</th>\n",
              "      <th>Chromosome</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>MT-RNR1</td>\n",
              "      <td>1517A&gt;C</td>\n",
              "      <td>1517</td>\n",
              "      <td>A</td>\n",
              "      <td>C</td>\n",
              "      <td>1518</td>\n",
              "      <td>M</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>MT-RNR1</td>\n",
              "      <td>1537C&gt;T</td>\n",
              "      <td>1537</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>1538</td>\n",
              "      <td>M</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>MT-RNR1</td>\n",
              "      <td>801A&gt;G</td>\n",
              "      <td>801</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>802</td>\n",
              "      <td>M</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>MT-TG</td>\n",
              "      <td>10003T&gt;C</td>\n",
              "      <td>10003</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>10004</td>\n",
              "      <td>M</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>MT-TG</td>\n",
              "      <td>10005A&gt;G</td>\n",
              "      <td>10005</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>10006</td>\n",
              "      <td>M</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>115</th>\n",
              "      <td>MT-RNR1</td>\n",
              "      <td>960dupC</td>\n",
              "      <td>960</td>\n",
              "      <td>C</td>\n",
              "      <td>CC</td>\n",
              "      <td>962</td>\n",
              "      <td>M</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>116</th>\n",
              "      <td>MT-RNR1</td>\n",
              "      <td>988G&gt;A</td>\n",
              "      <td>988</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>989</td>\n",
              "      <td>M</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>117</th>\n",
              "      <td>MT-RNR1</td>\n",
              "      <td>990T&gt;C</td>\n",
              "      <td>990</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>991</td>\n",
              "      <td>M</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>118</th>\n",
              "      <td>MT-TN</td>\n",
              "      <td>5715A&gt;G</td>\n",
              "      <td>5715</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>5716</td>\n",
              "      <td>M</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>119</th>\n",
              "      <td>MT-RNR1</td>\n",
              "      <td>669T&gt;C</td>\n",
              "      <td>669</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>670</td>\n",
              "      <td>M</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>120 rows Ã— 7 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "        Gene  Position  Start Ref Alt    End Chromosome\n",
              "0    MT-RNR1   1517A>C   1517   A   C   1518          M\n",
              "1    MT-RNR1   1537C>T   1537   C   T   1538          M\n",
              "2    MT-RNR1    801A>G    801   A   G    802          M\n",
              "3      MT-TG  10003T>C  10003   T   C  10004          M\n",
              "4      MT-TG  10005A>G  10005   A   G  10006          M\n",
              "..       ...       ...    ...  ..  ..    ...        ...\n",
              "115  MT-RNR1   960dupC    960   C  CC    962          M\n",
              "116  MT-RNR1    988G>A    988   G   A    989          M\n",
              "117  MT-RNR1    990T>C    990   T   C    991          M\n",
              "118    MT-TN   5715A>G   5715   A   G   5716          M\n",
              "119  MT-RNR1    669T>C    669   T   C    670          M\n",
              "\n",
              "[120 rows x 7 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 26
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 49
        },
        "id": "_--UE3iwoHGc",
        "outputId": "4c2d88b2-c394-4610-bfc4-fa3d59e443ae"
      },
      "source": [
        "## Checking overlap with clinar database\n",
        "df_clinvar_merge = pd.merge(df_clinvar, df_mt, left_on=['Chromosome','Start','End', 'Ref', 'Alt'], right_on=['Chromosome','Start','End', 'Ref', 'Alt'],)\n",
        "df_clinvar_merge"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "      <th>End</th>\n",
              "      <th>Ref</th>\n",
              "      <th>Alt</th>\n",
              "      <th>var_type</th>\n",
              "      <th>mol_consequence</th>\n",
              "      <th>Gene</th>\n",
              "      <th>Position</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "Empty DataFrame\n",
              "Columns: [Chromosome, Start, End, Ref, Alt, var_type, mol_consequence, Gene, Position]\n",
              "Index: []"
            ]
          },
          "metadata": {},
          "execution_count": 27
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "QIFy_c4WFRMd"
      },
      "source": [
        "### Merging variants - Genes and MT\n",
        "\n",
        "**results:**\n",
        "- Output file to use on filter script \n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "_tFdog2VFQ6U",
        "outputId": "da22018f-4e66-4f38-f5f6-4cb4b03c44a2"
      },
      "source": [
        "annotate_transcript_df = annotate_transcript.df\n",
        "annotate_transcript_df = annotate_transcript_df[['Chromosome', 'Start']]\n",
        "annotate_transcript_df"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>6425219</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>6425247</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>6445936</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>6448931</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1</td>\n",
              "      <td>6451602</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5415</th>\n",
              "      <td>X</td>\n",
              "      <td>130147792</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5416</th>\n",
              "      <td>X</td>\n",
              "      <td>130147804</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5417</th>\n",
              "      <td>X</td>\n",
              "      <td>130149478</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5418</th>\n",
              "      <td>X</td>\n",
              "      <td>130156576</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5419</th>\n",
              "      <td>X</td>\n",
              "      <td>130165779</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5420 rows Ã— 2 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "     Chromosome      Start\n",
              "0             1    6425219\n",
              "1             1    6425247\n",
              "2             1    6445936\n",
              "3             1    6448931\n",
              "4             1    6451602\n",
              "...         ...        ...\n",
              "5415          X  130147792\n",
              "5416          X  130147804\n",
              "5417          X  130149478\n",
              "5418          X  130156576\n",
              "5419          X  130165779\n",
              "\n",
              "[5420 rows x 2 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 28
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "ByP07NqYGIIu",
        "outputId": "d26c9429-0828-48b5-f2ad-c7ae56420867"
      },
      "source": [
        "df_mt.head()\n",
        "df_mt = df_mt[['Chromosome','Start']]\n",
        "df_mt"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>M</td>\n",
              "      <td>1517</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>M</td>\n",
              "      <td>1537</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>M</td>\n",
              "      <td>801</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>M</td>\n",
              "      <td>10003</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>M</td>\n",
              "      <td>10005</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>115</th>\n",
              "      <td>M</td>\n",
              "      <td>960</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>116</th>\n",
              "      <td>M</td>\n",
              "      <td>988</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>117</th>\n",
              "      <td>M</td>\n",
              "      <td>990</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>118</th>\n",
              "      <td>M</td>\n",
              "      <td>5715</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>119</th>\n",
              "      <td>M</td>\n",
              "      <td>669</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>120 rows Ã— 2 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "    Chromosome  Start\n",
              "0            M   1517\n",
              "1            M   1537\n",
              "2            M    801\n",
              "3            M  10003\n",
              "4            M  10005\n",
              "..         ...    ...\n",
              "115          M    960\n",
              "116          M    988\n",
              "117          M    990\n",
              "118          M   5715\n",
              "119          M    669\n",
              "\n",
              "[120 rows x 2 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 29
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "bOl-69k6IUs6",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "outputId": "a432a938-44e9-421c-b25c-a5b2ca950004"
      },
      "source": [
        "df_final = pd.concat([annotate_transcript_df, df_mt])\n",
        "df_final = df_final.drop_duplicates()\n",
        "df_final['Chromosome'] = df_final['Chromosome'].map('chr{}'.format)\n",
        "df_final\n",
        "#df_final.to_csv('/content/drive/MyDrive/GRAR/Paper_Surdez/All_variants_position.txt', index = None, header = False, sep = '\\t')"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr1</td>\n",
              "      <td>6425219</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr1</td>\n",
              "      <td>6425247</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr1</td>\n",
              "      <td>6445936</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr1</td>\n",
              "      <td>6448931</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr1</td>\n",
              "      <td>6451602</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>114</th>\n",
              "      <td>chrM</td>\n",
              "      <td>960</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>116</th>\n",
              "      <td>chrM</td>\n",
              "      <td>988</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>117</th>\n",
              "      <td>chrM</td>\n",
              "      <td>990</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>118</th>\n",
              "      <td>chrM</td>\n",
              "      <td>5715</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>119</th>\n",
              "      <td>chrM</td>\n",
              "      <td>669</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5330 rows Ã— 2 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "    Chromosome    Start\n",
              "0         chr1  6425219\n",
              "1         chr1  6425247\n",
              "2         chr1  6445936\n",
              "3         chr1  6448931\n",
              "4         chr1  6451602\n",
              "..         ...      ...\n",
              "114       chrM      960\n",
              "116       chrM      988\n",
              "117       chrM      990\n",
              "118       chrM     5715\n",
              "119       chrM      669\n",
              "\n",
              "[5330 rows x 2 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 30
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "3BHvbEWx9rsM"
      },
      "source": [
        "### Metrics to be considered between all filtered variants\n",
        "\n",
        "**results**\n",
        "- Quantity of each variant type\n",
        "- Quantity of molecular consequences within var type\n",
        "- Final variants - not includins genes FAM189A2, DFNX3, KCNJ10"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 142
        },
        "id": "cfJ-aMr1zXqK",
        "outputId": "933c3cb8-1e9c-4d4f-cf28-d13ebebdaa48"
      },
      "source": [
        "df_final = annotate_transcript.df\n",
        "df_final_genes = df_final['Gene'].drop_duplicates().copy()\n",
        "\n",
        "genes_diff = pd.merge(df_genes, df_final_genes, left_on='Genes', right_on='Gene', how='outer', indicator=True )\n",
        "genes_diff[genes_diff['_merge'] == 'left_only']"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Genes</th>\n",
              "      <th>Gene</th>\n",
              "      <th>_merge</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>DFNX3</td>\n",
              "      <td>NaN</td>\n",
              "      <td>left_only</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>36</th>\n",
              "      <td>FAM189A2</td>\n",
              "      <td>NaN</td>\n",
              "      <td>left_only</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>55</th>\n",
              "      <td>KCNJ10</td>\n",
              "      <td>NaN</td>\n",
              "      <td>left_only</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "       Genes Gene     _merge\n",
              "24     DFNX3  NaN  left_only\n",
              "36  FAM189A2  NaN  left_only\n",
              "55   KCNJ10   NaN  left_only"
            ]
          },
          "metadata": {},
          "execution_count": 31
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 266
        },
        "id": "8oc2zm-soVNf",
        "outputId": "055c9f3c-a1ae-4c02-dc15-ddad61bfc09c"
      },
      "source": [
        "\n",
        "df_final.groupby(by=['var_type']).size().reset_index()"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>var_type</th>\n",
              "      <th>0</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>Deletion</td>\n",
              "      <td>612</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>Duplication</td>\n",
              "      <td>225</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>Indel</td>\n",
              "      <td>42</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>Insertion</td>\n",
              "      <td>31</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>Inversion</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>Microsatellite</td>\n",
              "      <td>105</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>4359</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                    var_type     0\n",
              "0                   Deletion   612\n",
              "1                Duplication   225\n",
              "2                      Indel    42\n",
              "3                  Insertion    31\n",
              "4                  Inversion     1\n",
              "5             Microsatellite   105\n",
              "6  single_nucleotide_variant  4359"
            ]
          },
          "metadata": {},
          "execution_count": 32
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "M0hUFSY5qP-n",
        "outputId": "55cd8253-b952-4044-d6b4-e587b874c508"
      },
      "source": [
        "df_final.groupby(by=['var_type','mol_consequence']).size()"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "var_type                   mol_consequence                    \n",
              "Deletion                   frameshift_variant                      504\n",
              "                           genic_downstream_transcript_variant       2\n",
              "                           genic_upstream_transcript_variant         1\n",
              "                           inframe_deletion                         21\n",
              "                           inframe_indel                             2\n",
              "                           initiatior_codon_variant                  1\n",
              "                           intron_variant                           14\n",
              "                           non-coding_transcript_variant             5\n",
              "                           nonsense                                 22\n",
              "                           splice_acceptor_variant                  17\n",
              "                           splice_donor_variant                     22\n",
              "                           stop_lost                                 1\n",
              "Duplication                frameshift_variant                      186\n",
              "                           inframe_insertion                         6\n",
              "                           intron_variant                            6\n",
              "                           non-coding_transcript_variant             1\n",
              "                           nonsense                                 20\n",
              "                           splice_acceptor_variant                   2\n",
              "                           splice_donor_variant                      4\n",
              "Indel                      frameshift_variant                       23\n",
              "                           missense_variant                          6\n",
              "                           non-coding_transcript_variant             1\n",
              "                           nonsense                                  3\n",
              "                           splice_acceptor_variant                   6\n",
              "                           splice_donor_variant                      3\n",
              "Insertion                  frameshift_variant                       24\n",
              "                           intron_variant                            1\n",
              "                           non-coding_transcript_variant             1\n",
              "                           nonsense                                  5\n",
              "Inversion                  splice_acceptor_variant                   1\n",
              "Microsatellite             frameshift_variant                       55\n",
              "                           inframe_deletion                         32\n",
              "                           inframe_insertion                         2\n",
              "                           intron_variant                            9\n",
              "                           non-coding_transcript_variant             1\n",
              "                           nonsense                                  3\n",
              "                           splice_donor_variant                      3\n",
              "single_nucleotide_variant  3_prime_UTR_variant                       3\n",
              "                           5_prime_UTR_variant                       1\n",
              "                           initiatior_codon_variant                 13\n",
              "                           intron_variant                          272\n",
              "                           missense_variant                       2068\n",
              "                           non-coding_transcript_variant            93\n",
              "                           nonsense                                739\n",
              "                           splice_acceptor_variant                 197\n",
              "                           splice_donor_variant                    269\n",
              "                           stop_lost                                 5\n",
              "                           synonymous_variant                      699\n",
              "dtype: int64"
            ]
          },
          "metadata": {},
          "execution_count": 33
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Nkx1Doddn1hS"
      },
      "source": [
        "### Annotating variants after merge with clinvar"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "ssH7TDPpopIt",
        "outputId": "743e195e-e329-42e4-e773-5b1030cde6df"
      },
      "source": [
        "df_after_merge = pd.read_table('/content/drive/MyDrive/GRAR/Paper_Surdez/gene_names_alleles.txt', sep ='\\t')\n",
        "df_after_merge['chr'] = df_after_merge['chr'].str.replace('chr','')\n",
        "#df_after_merge['chr'] = df_after_merge['chr'].str.replace('MT','M')\n",
        "df_after_merge"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>chr</th>\n",
              "      <th>position</th>\n",
              "      <th>gene_names</th>\n",
              "      <th>ref</th>\n",
              "      <th>alt</th>\n",
              "      <th>alt_list</th>\n",
              "      <th>var_type</th>\n",
              "      <th>mol_consequence</th>\n",
              "      <th>biallelic</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>6425219</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "      <td>yes</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>6425247</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "      <td>yes</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>6445936</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>splice_donor_variant</td>\n",
              "      <td>yes</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>6448931</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>C</td>\n",
              "      <td>CG</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Duplication</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>yes</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1</td>\n",
              "      <td>6451602</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>G</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>splice_acceptor_variant</td>\n",
              "      <td>yes</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5325</th>\n",
              "      <td>MT</td>\n",
              "      <td>15927</td>\n",
              "      <td>MT-TT</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5326</th>\n",
              "      <td>MT</td>\n",
              "      <td>15975</td>\n",
              "      <td>MT-TP</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5327</th>\n",
              "      <td>MT</td>\n",
              "      <td>15992</td>\n",
              "      <td>MT-TP</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5328</th>\n",
              "      <td>MT</td>\n",
              "      <td>15997</td>\n",
              "      <td>MT-TP</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5329</th>\n",
              "      <td>MT</td>\n",
              "      <td>16023</td>\n",
              "      <td>MT-TP</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5330 rows Ã— 9 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "     chr  position  ...          mol_consequence biallelic\n",
              "0      1   6425219  ...                 nonsense       yes\n",
              "1      1   6425247  ...                 nonsense       yes\n",
              "2      1   6445936  ...     splice_donor_variant       yes\n",
              "3      1   6448931  ...       frameshift_variant       yes\n",
              "4      1   6451602  ...  splice_acceptor_variant       yes\n",
              "...   ..       ...  ...                      ...       ...\n",
              "5325  MT     15927  ...                      NaN       yes\n",
              "5326  MT     15975  ...                      NaN       yes\n",
              "5327  MT     15992  ...                      NaN       yes\n",
              "5328  MT     15997  ...                      NaN       yes\n",
              "5329  MT     16023  ...                      NaN       yes\n",
              "\n",
              "[5330 rows x 9 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 34
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "CVTu2h7eo6bK",
        "outputId": "4e2a25e3-6be4-4551-eeea-5b868b922311"
      },
      "source": [
        "clinsig_regex = r'.*CLNSIG=(.*?);.*' \n",
        "#CLNSIG=Pathogenic|CLNSIG=Likely_pathogenic|CLNSIG=Conflicting_interpretations_of_pathogenicity \n",
        "df_clinvar_total['pathogenicity'] = df_clinvar_total[7].str.extract(clinsig_regex, expand=True)\n",
        "df_clinvar_total = df_clinvar_total[[0,1,3,4,'pathogenicity']]\n",
        "df_clinvar_total"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>3</th>\n",
              "      <th>4</th>\n",
              "      <th>pathogenicity</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>925952</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>Uncertain_significance</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>930139</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>Likely_benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>930165</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>930187</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>Likely_benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1</td>\n",
              "      <td>930188</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>Uncertain_significance</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083268</th>\n",
              "      <td>MT</td>\n",
              "      <td>16013</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>Uncertain_significance</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083269</th>\n",
              "      <td>MT</td>\n",
              "      <td>16015</td>\n",
              "      <td>T</td>\n",
              "      <td>TATTCTCTGTTCTTTC</td>\n",
              "      <td>Uncertain_significance</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083270</th>\n",
              "      <td>MT</td>\n",
              "      <td>16017</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083271</th>\n",
              "      <td>MT</td>\n",
              "      <td>16021</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>Uncertain_significance</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083272</th>\n",
              "      <td>NW_009646201.1</td>\n",
              "      <td>83614</td>\n",
              "      <td>TC</td>\n",
              "      <td>T</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>1083273 rows Ã— 5 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "                      0       1   3                 4           pathogenicity\n",
              "0                     1  925952   G                 A  Uncertain_significance\n",
              "1                     1  930139   C                 T           Likely_benign\n",
              "2                     1  930165   G                 A                  Benign\n",
              "3                     1  930187   C                 T           Likely_benign\n",
              "4                     1  930188   G                 A  Uncertain_significance\n",
              "...                 ...     ...  ..               ...                     ...\n",
              "1083268              MT   16013   A                 G  Uncertain_significance\n",
              "1083269              MT   16015   T  TATTCTCTGTTCTTTC  Uncertain_significance\n",
              "1083270              MT   16017   T                 C                  Benign\n",
              "1083271              MT   16021   C                 T  Uncertain_significance\n",
              "1083272  NW_009646201.1   83614  TC                 T                  Benign\n",
              "\n",
              "[1083273 rows x 5 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 35
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "fdC8RhAbx35O",
        "outputId": "c8d2a9ba-0677-4e83-f590-3a6b8f0f7a83"
      },
      "source": [
        "df_clinvar_total[df_clinvar_total['pathogenicity'].isnull()]"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>3</th>\n",
              "      <th>4</th>\n",
              "      <th>pathogenicity</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>6481</th>\n",
              "      <td>1</td>\n",
              "      <td>7984971</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7132</th>\n",
              "      <td>1</td>\n",
              "      <td>9975751</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17540</th>\n",
              "      <td>1</td>\n",
              "      <td>25284610</td>\n",
              "      <td>G</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17545</th>\n",
              "      <td>1</td>\n",
              "      <td>25290715</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17547</th>\n",
              "      <td>1</td>\n",
              "      <td>25290760</td>\n",
              "      <td>A</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1076266</th>\n",
              "      <td>X</td>\n",
              "      <td>154030761</td>\n",
              "      <td>CTGCTGCGCCCCTTGGGGCTGCTCTCCT</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1076287</th>\n",
              "      <td>X</td>\n",
              "      <td>154030791</td>\n",
              "      <td>CTTTTCCGCCCAGGGCTCTTACAGGTCT</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1076313</th>\n",
              "      <td>X</td>\n",
              "      <td>154030839</td>\n",
              "      <td>TCA</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1079407</th>\n",
              "      <td>X</td>\n",
              "      <td>154534438</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1079436</th>\n",
              "      <td>X</td>\n",
              "      <td>154535336</td>\n",
              "      <td>G</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>612 rows Ã— 5 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "         0          1                             3  4 pathogenicity\n",
              "6481     1    7984971                             G  A           NaN\n",
              "7132     1    9975751                             G  A           NaN\n",
              "17540    1   25284610                             G  T           NaN\n",
              "17545    1   25290715                             C  T           NaN\n",
              "17547    1   25290760                             A  C           NaN\n",
              "...     ..        ...                           ... ..           ...\n",
              "1076266  X  154030761  CTGCTGCGCCCCTTGGGGCTGCTCTCCT  C           NaN\n",
              "1076287  X  154030791  CTTTTCCGCCCAGGGCTCTTACAGGTCT  C           NaN\n",
              "1076313  X  154030839                           TCA  T           NaN\n",
              "1079407  X  154534438                             G  A           NaN\n",
              "1079436  X  154535336                             G  C           NaN\n",
              "\n",
              "[612 rows x 5 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 36
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "nrCeHOy6zHs4",
        "outputId": "8b6e4e3a-c707-491c-cc0c-c6db84fa35e1"
      },
      "source": [
        "df_clinvar_total.columns = ['chr','position','ref','alt','pathogenicity']\n",
        "df_clinvar_total"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>chr</th>\n",
              "      <th>position</th>\n",
              "      <th>ref</th>\n",
              "      <th>alt</th>\n",
              "      <th>pathogenicity</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>925952</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>Uncertain_significance</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>930139</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>Likely_benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>930165</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>930187</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>Likely_benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1</td>\n",
              "      <td>930188</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>Uncertain_significance</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083268</th>\n",
              "      <td>MT</td>\n",
              "      <td>16013</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>Uncertain_significance</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083269</th>\n",
              "      <td>MT</td>\n",
              "      <td>16015</td>\n",
              "      <td>T</td>\n",
              "      <td>TATTCTCTGTTCTTTC</td>\n",
              "      <td>Uncertain_significance</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083270</th>\n",
              "      <td>MT</td>\n",
              "      <td>16017</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083271</th>\n",
              "      <td>MT</td>\n",
              "      <td>16021</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>Uncertain_significance</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1083272</th>\n",
              "      <td>NW_009646201.1</td>\n",
              "      <td>83614</td>\n",
              "      <td>TC</td>\n",
              "      <td>T</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>1083273 rows Ã— 5 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "                    chr  position ref               alt           pathogenicity\n",
              "0                     1    925952   G                 A  Uncertain_significance\n",
              "1                     1    930139   C                 T           Likely_benign\n",
              "2                     1    930165   G                 A                  Benign\n",
              "3                     1    930187   C                 T           Likely_benign\n",
              "4                     1    930188   G                 A  Uncertain_significance\n",
              "...                 ...       ...  ..               ...                     ...\n",
              "1083268              MT     16013   A                 G  Uncertain_significance\n",
              "1083269              MT     16015   T  TATTCTCTGTTCTTTC  Uncertain_significance\n",
              "1083270              MT     16017   T                 C                  Benign\n",
              "1083271              MT     16021   C                 T  Uncertain_significance\n",
              "1083272  NW_009646201.1     83614  TC                 T                  Benign\n",
              "\n",
              "[1083273 rows x 5 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 37
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "OLn9HUFZ5tEr",
        "outputId": "1ef52a29-d2a7-45b3-e294-98eac91d4717"
      },
      "source": [
        "df_after_merge_annotate = pd.merge(df_after_merge, df_clinvar_total, left_on=['chr','position','ref', 'alt'], right_on=['chr','position','ref', 'alt'], how='left' )\n",
        "df_after_merge_annotate"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>chr</th>\n",
              "      <th>position</th>\n",
              "      <th>gene_names</th>\n",
              "      <th>ref</th>\n",
              "      <th>alt</th>\n",
              "      <th>alt_list</th>\n",
              "      <th>var_type</th>\n",
              "      <th>mol_consequence</th>\n",
              "      <th>biallelic</th>\n",
              "      <th>pathogenicity</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>6425219</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "      <td>yes</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>6425247</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "      <td>yes</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>6445936</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>splice_donor_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>6448931</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>C</td>\n",
              "      <td>CG</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Duplication</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1</td>\n",
              "      <td>6451602</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>G</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>splice_acceptor_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5325</th>\n",
              "      <td>MT</td>\n",
              "      <td>15927</td>\n",
              "      <td>MT-TT</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5326</th>\n",
              "      <td>MT</td>\n",
              "      <td>15975</td>\n",
              "      <td>MT-TP</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5327</th>\n",
              "      <td>MT</td>\n",
              "      <td>15992</td>\n",
              "      <td>MT-TP</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5328</th>\n",
              "      <td>MT</td>\n",
              "      <td>15997</td>\n",
              "      <td>MT-TP</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5329</th>\n",
              "      <td>MT</td>\n",
              "      <td>16023</td>\n",
              "      <td>MT-TP</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5330 rows Ã— 10 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "     chr  position gene_names  ...          mol_consequence biallelic pathogenicity\n",
              "0      1   6425219       ESPN  ...                 nonsense       yes           NaN\n",
              "1      1   6425247       ESPN  ...                 nonsense       yes           NaN\n",
              "2      1   6445936       ESPN  ...     splice_donor_variant       yes           NaN\n",
              "3      1   6448931       ESPN  ...       frameshift_variant       yes           NaN\n",
              "4      1   6451602       ESPN  ...  splice_acceptor_variant       yes           NaN\n",
              "...   ..       ...        ...  ...                      ...       ...           ...\n",
              "5325  MT     15927      MT-TT  ...                      NaN       yes        Benign\n",
              "5326  MT     15975      MT-TP  ...                      NaN       yes           NaN\n",
              "5327  MT     15992      MT-TP  ...                      NaN       yes           NaN\n",
              "5328  MT     15997      MT-TP  ...                      NaN       yes           NaN\n",
              "5329  MT     16023      MT-TP  ...                      NaN       yes           NaN\n",
              "\n",
              "[5330 rows x 10 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 38
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "wV-hyETj9mOk"
      },
      "source": [
        "df_after_merge_annotate.to_csv('/content/drive/MyDrive/GRAR/Paper_Surdez/Variants_after_merge_annotated.txt', index = None, header = True, sep = '\\t')"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "enLuexFu6Qon",
        "outputId": "06fa8879-6209-42e1-be78-4a83ef42e97d"
      },
      "source": [
        "df_after_merge_annotate[~df_after_merge_annotate['pathogenicity'].isnull()]"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>chr</th>\n",
              "      <th>position</th>\n",
              "      <th>gene_names</th>\n",
              "      <th>ref</th>\n",
              "      <th>alt</th>\n",
              "      <th>alt_list</th>\n",
              "      <th>var_type</th>\n",
              "      <th>mol_consequence</th>\n",
              "      <th>biallelic</th>\n",
              "      <th>pathogenicity</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>4847</th>\n",
              "      <td>20</td>\n",
              "      <td>62260083</td>\n",
              "      <td>OSBPL2</td>\n",
              "      <td>CTG</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Deletion</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4848</th>\n",
              "      <td>20</td>\n",
              "      <td>62260095</td>\n",
              "      <td>OSBPL2</td>\n",
              "      <td>CCT</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Microsatellite</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4849</th>\n",
              "      <td>20</td>\n",
              "      <td>62260100</td>\n",
              "      <td>OSBPL2</td>\n",
              "      <td>CAA</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Deletion</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4850</th>\n",
              "      <td>20</td>\n",
              "      <td>62260119</td>\n",
              "      <td>OSBPL2</td>\n",
              "      <td>AAC</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Microsatellite</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4851</th>\n",
              "      <td>21</td>\n",
              "      <td>36461002</td>\n",
              "      <td>CLDN14</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>Pathogenic/Likely_pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5319</th>\n",
              "      <td>MT</td>\n",
              "      <td>14693</td>\n",
              "      <td>MT-TE</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5321</th>\n",
              "      <td>MT</td>\n",
              "      <td>15902</td>\n",
              "      <td>MT-TT</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5322</th>\n",
              "      <td>MT</td>\n",
              "      <td>15908</td>\n",
              "      <td>MT-TT</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5324</th>\n",
              "      <td>MT</td>\n",
              "      <td>15926</td>\n",
              "      <td>MT-TT</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5325</th>\n",
              "      <td>MT</td>\n",
              "      <td>15927</td>\n",
              "      <td>MT-TT</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>428 rows Ã— 10 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "     chr  position  ... biallelic                 pathogenicity\n",
              "4847  20  62260083  ...       yes                    Pathogenic\n",
              "4848  20  62260095  ...       yes                    Pathogenic\n",
              "4849  20  62260100  ...       yes                    Pathogenic\n",
              "4850  20  62260119  ...       yes                    Pathogenic\n",
              "4851  21  36461002  ...       yes  Pathogenic/Likely_pathogenic\n",
              "...   ..       ...  ...       ...                           ...\n",
              "5319  MT     14693  ...       yes                        Benign\n",
              "5321  MT     15902  ...       yes                        Benign\n",
              "5322  MT     15908  ...       yes                        Benign\n",
              "5324  MT     15926  ...       yes                        Benign\n",
              "5325  MT     15927  ...       yes                        Benign\n",
              "\n",
              "[428 rows x 10 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 40
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "7OE3n-KY6Rwx",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "7f30ea73-d33c-46fb-b0b7-a3165e6c1d04"
      },
      "source": [
        "df_variants_summary = pd.read_table('/content/drive/MyDrive/GRAR/Paper_Surdez/variant_summary.txt', sep ='\\t')"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2822: DtypeWarning: Columns (18) have mixed types.Specify dtype option on import or set low_memory=False.\n",
            "  if self.run_code(code, result):\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 796
        },
        "id": "5om0aO1yu8XD",
        "outputId": "a1458be4-9101-4f61-b7a3-f6cbe11496f9"
      },
      "source": [
        "df_variants_summary = df_variants_summary[df_variants_summary['Assembly'] == 'GRCh38']\n",
        "df_variants_summary"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>#AlleleID</th>\n",
              "      <th>Type</th>\n",
              "      <th>Name</th>\n",
              "      <th>GeneID</th>\n",
              "      <th>GeneSymbol</th>\n",
              "      <th>HGNC_ID</th>\n",
              "      <th>ClinicalSignificance</th>\n",
              "      <th>ClinSigSimple</th>\n",
              "      <th>LastEvaluated</th>\n",
              "      <th>RS# (dbSNP)</th>\n",
              "      <th>nsv/esv (dbVar)</th>\n",
              "      <th>RCVaccession</th>\n",
              "      <th>PhenotypeIDS</th>\n",
              "      <th>PhenotypeList</th>\n",
              "      <th>Origin</th>\n",
              "      <th>OriginSimple</th>\n",
              "      <th>Assembly</th>\n",
              "      <th>ChromosomeAccession</th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "      <th>Stop</th>\n",
              "      <th>ReferenceAllele</th>\n",
              "      <th>AlternateAllele</th>\n",
              "      <th>Cytogenetic</th>\n",
              "      <th>ReviewStatus</th>\n",
              "      <th>NumberSubmitters</th>\n",
              "      <th>Guidelines</th>\n",
              "      <th>TestedInGTR</th>\n",
              "      <th>OtherIDs</th>\n",
              "      <th>SubmitterCategories</th>\n",
              "      <th>VariationID</th>\n",
              "      <th>PositionVCF</th>\n",
              "      <th>ReferenceAlleleVCF</th>\n",
              "      <th>AlternateAlleleVCF</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>15041</td>\n",
              "      <td>Indel</td>\n",
              "      <td>NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...</td>\n",
              "      <td>9907</td>\n",
              "      <td>AP5Z1</td>\n",
              "      <td>HGNC:22197</td>\n",
              "      <td>Pathogenic</td>\n",
              "      <td>1</td>\n",
              "      <td>-</td>\n",
              "      <td>397704705</td>\n",
              "      <td>-</td>\n",
              "      <td>RCV000000012</td>\n",
              "      <td>MONDO:MONDO:0013342,MedGen:C3150901,OMIM:61364...</td>\n",
              "      <td>Spastic paraplegia 48, autosomal recessive</td>\n",
              "      <td>germline;unknown</td>\n",
              "      <td>germline</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>NC_000007.14</td>\n",
              "      <td>7</td>\n",
              "      <td>4781213</td>\n",
              "      <td>4781216</td>\n",
              "      <td>na</td>\n",
              "      <td>na</td>\n",
              "      <td>7p22.1</td>\n",
              "      <td>criteria provided, single submitter</td>\n",
              "      <td>2</td>\n",
              "      <td>-</td>\n",
              "      <td>N</td>\n",
              "      <td>ClinGen:CA215070,OMIM:613653.0001</td>\n",
              "      <td>3</td>\n",
              "      <td>2</td>\n",
              "      <td>4781213</td>\n",
              "      <td>GGAT</td>\n",
              "      <td>TGCTGTAAACTGTAACTGTAAA</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>15042</td>\n",
              "      <td>Deletion</td>\n",
              "      <td>NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)</td>\n",
              "      <td>9907</td>\n",
              "      <td>AP5Z1</td>\n",
              "      <td>HGNC:22197</td>\n",
              "      <td>Pathogenic</td>\n",
              "      <td>1</td>\n",
              "      <td>Jun 29, 2010</td>\n",
              "      <td>397704709</td>\n",
              "      <td>-</td>\n",
              "      <td>RCV000000013</td>\n",
              "      <td>MONDO:MONDO:0013342,MedGen:C3150901,OMIM:61364...</td>\n",
              "      <td>Spastic paraplegia 48, autosomal recessive</td>\n",
              "      <td>germline</td>\n",
              "      <td>germline</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>NC_000007.14</td>\n",
              "      <td>7</td>\n",
              "      <td>4787730</td>\n",
              "      <td>4787743</td>\n",
              "      <td>na</td>\n",
              "      <td>na</td>\n",
              "      <td>7p22.1</td>\n",
              "      <td>no assertion criteria provided</td>\n",
              "      <td>1</td>\n",
              "      <td>-</td>\n",
              "      <td>N</td>\n",
              "      <td>ClinGen:CA215072,OMIM:613653.0002</td>\n",
              "      <td>1</td>\n",
              "      <td>3</td>\n",
              "      <td>4787729</td>\n",
              "      <td>GCTGCTGGACCTGCC</td>\n",
              "      <td>G</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>15043</td>\n",
              "      <td>single nucleotide variant</td>\n",
              "      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n",
              "      <td>9640</td>\n",
              "      <td>ZNF592</td>\n",
              "      <td>HGNC:28986</td>\n",
              "      <td>Uncertain significance</td>\n",
              "      <td>0</td>\n",
              "      <td>Jun 29, 2015</td>\n",
              "      <td>150829393</td>\n",
              "      <td>-</td>\n",
              "      <td>RCV000000014</td>\n",
              "      <td>MONDO:MONDO:0033005,MedGen:C4551772,OMIM:25130...</td>\n",
              "      <td>Galloway-Mowat syndrome 1</td>\n",
              "      <td>germline</td>\n",
              "      <td>germline</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>NC_000015.10</td>\n",
              "      <td>15</td>\n",
              "      <td>84799209</td>\n",
              "      <td>84799209</td>\n",
              "      <td>na</td>\n",
              "      <td>na</td>\n",
              "      <td>15q25.3</td>\n",
              "      <td>no assertion criteria provided</td>\n",
              "      <td>1</td>\n",
              "      <td>-</td>\n",
              "      <td>N</td>\n",
              "      <td>ClinGen:CA210674,UniProtKB:Q92610#VAR_064583,O...</td>\n",
              "      <td>1</td>\n",
              "      <td>4</td>\n",
              "      <td>84799209</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>15044</td>\n",
              "      <td>single nucleotide variant</td>\n",
              "      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n",
              "      <td>55572</td>\n",
              "      <td>FOXRED1</td>\n",
              "      <td>HGNC:26927</td>\n",
              "      <td>Pathogenic</td>\n",
              "      <td>1</td>\n",
              "      <td>Dec 30, 2019</td>\n",
              "      <td>267606829</td>\n",
              "      <td>-</td>\n",
              "      <td>RCV000578659|RCV001194045|RCV000000015</td>\n",
              "      <td>MedGen:CN517202|MONDO:MONDO:0009723,MedGen:C00...</td>\n",
              "      <td>not provided|Leigh syndrome|Mitochondrial comp...</td>\n",
              "      <td>germline</td>\n",
              "      <td>germline</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>NC_000011.10</td>\n",
              "      <td>11</td>\n",
              "      <td>126275389</td>\n",
              "      <td>126275389</td>\n",
              "      <td>na</td>\n",
              "      <td>na</td>\n",
              "      <td>11q24.2</td>\n",
              "      <td>criteria provided, multiple submitters, no con...</td>\n",
              "      <td>3</td>\n",
              "      <td>-</td>\n",
              "      <td>N</td>\n",
              "      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n",
              "      <td>3</td>\n",
              "      <td>5</td>\n",
              "      <td>126275389</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>15045</td>\n",
              "      <td>single nucleotide variant</td>\n",
              "      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n",
              "      <td>55572</td>\n",
              "      <td>FOXRED1</td>\n",
              "      <td>HGNC:26927</td>\n",
              "      <td>Pathogenic</td>\n",
              "      <td>1</td>\n",
              "      <td>Oct 01, 2010</td>\n",
              "      <td>267606830</td>\n",
              "      <td>-</td>\n",
              "      <td>RCV000000016</td>\n",
              "      <td>MONDO:MONDO:0032624,MedGen:C4748791,OMIM:618241</td>\n",
              "      <td>Mitochondrial complex 1 deficiency, nuclear ty...</td>\n",
              "      <td>germline</td>\n",
              "      <td>germline</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>NC_000011.10</td>\n",
              "      <td>11</td>\n",
              "      <td>126277517</td>\n",
              "      <td>126277517</td>\n",
              "      <td>na</td>\n",
              "      <td>na</td>\n",
              "      <td>11q24.2</td>\n",
              "      <td>no assertion criteria provided</td>\n",
              "      <td>1</td>\n",
              "      <td>-</td>\n",
              "      <td>N</td>\n",
              "      <td>UniProtKB:Q96CU9#VAR_064571,OMIM:613622.0002,C...</td>\n",
              "      <td>1</td>\n",
              "      <td>6</td>\n",
              "      <td>126277517</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2262902</th>\n",
              "      <td>1290058</td>\n",
              "      <td>single nucleotide variant</td>\n",
              "      <td>NM_001270974.2(HYDIN):c.2974G&gt;A (p.Ala992Thr)</td>\n",
              "      <td>54768</td>\n",
              "      <td>HYDIN</td>\n",
              "      <td>HGNC:19368</td>\n",
              "      <td>Likely benign</td>\n",
              "      <td>0</td>\n",
              "      <td>-</td>\n",
              "      <td>-1</td>\n",
              "      <td>-</td>\n",
              "      <td>RCV001730459</td>\n",
              "      <td>MedGen:CN517202</td>\n",
              "      <td>not provided</td>\n",
              "      <td>germline</td>\n",
              "      <td>germline</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>NC_000016.10</td>\n",
              "      <td>16</td>\n",
              "      <td>71027670</td>\n",
              "      <td>71027670</td>\n",
              "      <td>na</td>\n",
              "      <td>na</td>\n",
              "      <td>16q22.2</td>\n",
              "      <td>no assertion criteria provided</td>\n",
              "      <td>2</td>\n",
              "      <td>-</td>\n",
              "      <td>N</td>\n",
              "      <td>-</td>\n",
              "      <td>2</td>\n",
              "      <td>1300007</td>\n",
              "      <td>71027670</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2262904</th>\n",
              "      <td>1290059</td>\n",
              "      <td>single nucleotide variant</td>\n",
              "      <td>NM_001324144.2(ZNF41):c.824A&gt;T (p.Glu275Val)</td>\n",
              "      <td>7592</td>\n",
              "      <td>ZNF41</td>\n",
              "      <td>HGNC:13107</td>\n",
              "      <td>Likely benign</td>\n",
              "      <td>0</td>\n",
              "      <td>-</td>\n",
              "      <td>-1</td>\n",
              "      <td>-</td>\n",
              "      <td>RCV001730460</td>\n",
              "      <td>MedGen:CN517202</td>\n",
              "      <td>not provided</td>\n",
              "      <td>germline</td>\n",
              "      <td>germline</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>NC_000023.11</td>\n",
              "      <td>X</td>\n",
              "      <td>47448946</td>\n",
              "      <td>47448946</td>\n",
              "      <td>na</td>\n",
              "      <td>na</td>\n",
              "      <td>Xp11.3</td>\n",
              "      <td>no assertion criteria provided</td>\n",
              "      <td>2</td>\n",
              "      <td>-</td>\n",
              "      <td>N</td>\n",
              "      <td>-</td>\n",
              "      <td>2</td>\n",
              "      <td>1300008</td>\n",
              "      <td>47448946</td>\n",
              "      <td>T</td>\n",
              "      <td>A</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2262906</th>\n",
              "      <td>1290060</td>\n",
              "      <td>single nucleotide variant</td>\n",
              "      <td>NM_015335.5(MED13L):c.1654A&gt;G (p.Ile552Val)</td>\n",
              "      <td>23389</td>\n",
              "      <td>MED13L</td>\n",
              "      <td>HGNC:22962</td>\n",
              "      <td>Likely benign</td>\n",
              "      <td>0</td>\n",
              "      <td>-</td>\n",
              "      <td>-1</td>\n",
              "      <td>-</td>\n",
              "      <td>RCV001730461</td>\n",
              "      <td>MedGen:CN517202</td>\n",
              "      <td>not provided</td>\n",
              "      <td>germline</td>\n",
              "      <td>germline</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>NC_000012.12</td>\n",
              "      <td>12</td>\n",
              "      <td>116008759</td>\n",
              "      <td>116008759</td>\n",
              "      <td>na</td>\n",
              "      <td>na</td>\n",
              "      <td>12q24.21</td>\n",
              "      <td>no assertion criteria provided</td>\n",
              "      <td>2</td>\n",
              "      <td>-</td>\n",
              "      <td>N</td>\n",
              "      <td>-</td>\n",
              "      <td>2</td>\n",
              "      <td>1300009</td>\n",
              "      <td>116008759</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2262908</th>\n",
              "      <td>1290061</td>\n",
              "      <td>Microsatellite</td>\n",
              "      <td>NM_001348716.2(KDM6B):c.753ACC[12] (p.Pro264del)</td>\n",
              "      <td>23135</td>\n",
              "      <td>KDM6B</td>\n",
              "      <td>HGNC:29012</td>\n",
              "      <td>Benign/Likely benign</td>\n",
              "      <td>0</td>\n",
              "      <td>-</td>\n",
              "      <td>-1</td>\n",
              "      <td>-</td>\n",
              "      <td>RCV001730462|RCV001730463</td>\n",
              "      <td>MedGen:CN517202|MedGen:CN169374</td>\n",
              "      <td>not provided|not specified</td>\n",
              "      <td>germline</td>\n",
              "      <td>germline</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>NC_000017.11</td>\n",
              "      <td>17</td>\n",
              "      <td>7846860</td>\n",
              "      <td>7846862</td>\n",
              "      <td>na</td>\n",
              "      <td>na</td>\n",
              "      <td>17p13.1</td>\n",
              "      <td>no assertion criteria provided</td>\n",
              "      <td>2</td>\n",
              "      <td>-</td>\n",
              "      <td>N</td>\n",
              "      <td>-</td>\n",
              "      <td>2</td>\n",
              "      <td>1300010</td>\n",
              "      <td>7846859</td>\n",
              "      <td>TACC</td>\n",
              "      <td>T</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2262910</th>\n",
              "      <td>1290062</td>\n",
              "      <td>single nucleotide variant</td>\n",
              "      <td>NM_032796.4(SYAP1):c.743C&gt;T (p.Thr248Met)</td>\n",
              "      <td>94056</td>\n",
              "      <td>SYAP1</td>\n",
              "      <td>HGNC:16273</td>\n",
              "      <td>Uncertain significance</td>\n",
              "      <td>0</td>\n",
              "      <td>-</td>\n",
              "      <td>-1</td>\n",
              "      <td>-</td>\n",
              "      <td>RCV001730464</td>\n",
              "      <td>MedGen:CN517202</td>\n",
              "      <td>not provided</td>\n",
              "      <td>germline</td>\n",
              "      <td>germline</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>NC_000023.11</td>\n",
              "      <td>X</td>\n",
              "      <td>16756681</td>\n",
              "      <td>16756681</td>\n",
              "      <td>na</td>\n",
              "      <td>na</td>\n",
              "      <td>Xp22.2</td>\n",
              "      <td>no assertion criteria provided</td>\n",
              "      <td>2</td>\n",
              "      <td>-</td>\n",
              "      <td>N</td>\n",
              "      <td>-</td>\n",
              "      <td>2</td>\n",
              "      <td>1300011</td>\n",
              "      <td>16756681</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>1104046 rows Ã— 34 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "         #AlleleID  ...      AlternateAlleleVCF\n",
              "1            15041  ...  TGCTGTAAACTGTAACTGTAAA\n",
              "3            15042  ...                       G\n",
              "5            15043  ...                       A\n",
              "7            15044  ...                       T\n",
              "9            15045  ...                       G\n",
              "...            ...  ...                     ...\n",
              "2262902    1290058  ...                       T\n",
              "2262904    1290059  ...                       A\n",
              "2262906    1290060  ...                       C\n",
              "2262908    1290061  ...                       T\n",
              "2262910    1290062  ...                       T\n",
              "\n",
              "[1104046 rows x 34 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 42
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "gKo3IwJ3xOYL",
        "outputId": "070fc9c0-d5c0-4a67-e1c7-61aff86227b8"
      },
      "source": [
        "df_variants_summary = df_variants_summary[['Chromosome','Start','ReferenceAlleleVCF','AlternateAlleleVCF','ClinicalSignificance']]\n",
        "df_variants_summary"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "      <th>ReferenceAlleleVCF</th>\n",
              "      <th>AlternateAlleleVCF</th>\n",
              "      <th>ClinicalSignificance</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>7</td>\n",
              "      <td>4781213</td>\n",
              "      <td>GGAT</td>\n",
              "      <td>TGCTGTAAACTGTAACTGTAAA</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>7</td>\n",
              "      <td>4787730</td>\n",
              "      <td>GCTGCTGGACCTGCC</td>\n",
              "      <td>G</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>15</td>\n",
              "      <td>84799209</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>Uncertain significance</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>11</td>\n",
              "      <td>126275389</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>11</td>\n",
              "      <td>126277517</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2262902</th>\n",
              "      <td>16</td>\n",
              "      <td>71027670</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>Likely benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2262904</th>\n",
              "      <td>X</td>\n",
              "      <td>47448946</td>\n",
              "      <td>T</td>\n",
              "      <td>A</td>\n",
              "      <td>Likely benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2262906</th>\n",
              "      <td>12</td>\n",
              "      <td>116008759</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>Likely benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2262908</th>\n",
              "      <td>17</td>\n",
              "      <td>7846860</td>\n",
              "      <td>TACC</td>\n",
              "      <td>T</td>\n",
              "      <td>Benign/Likely benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2262910</th>\n",
              "      <td>X</td>\n",
              "      <td>16756681</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>Uncertain significance</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>1104046 rows Ã— 5 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "        Chromosome      Start  ...      AlternateAlleleVCF    ClinicalSignificance\n",
              "1                7    4781213  ...  TGCTGTAAACTGTAACTGTAAA              Pathogenic\n",
              "3                7    4787730  ...                       G              Pathogenic\n",
              "5               15   84799209  ...                       A  Uncertain significance\n",
              "7               11  126275389  ...                       T              Pathogenic\n",
              "9               11  126277517  ...                       G              Pathogenic\n",
              "...            ...        ...  ...                     ...                     ...\n",
              "2262902         16   71027670  ...                       T           Likely benign\n",
              "2262904          X   47448946  ...                       A           Likely benign\n",
              "2262906         12  116008759  ...                       C           Likely benign\n",
              "2262908         17    7846860  ...                       T    Benign/Likely benign\n",
              "2262910          X   16756681  ...                       T  Uncertain significance\n",
              "\n",
              "[1104046 rows x 5 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 43
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "Qi2Z4ADXxWve"
      },
      "source": [
        "df_variants_summary.columns = ['chr','position','ref','alt','pathogenicity']"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "GiTATsrdz-gH",
        "outputId": "57355ae9-7c94-410d-bc08-b959b3f69fc3"
      },
      "source": [
        "df_after_merge_annotate_summary = pd.merge(df_after_merge, df_variants_summary, how='left' )\n",
        "df_after_merge_annotate_summary"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>chr</th>\n",
              "      <th>position</th>\n",
              "      <th>gene_names</th>\n",
              "      <th>ref</th>\n",
              "      <th>alt</th>\n",
              "      <th>alt_list</th>\n",
              "      <th>var_type</th>\n",
              "      <th>mol_consequence</th>\n",
              "      <th>biallelic</th>\n",
              "      <th>pathogenicity</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>6425219</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "      <td>yes</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>6425247</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "      <td>yes</td>\n",
              "      <td>Likely pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>6445936</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>splice_donor_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>Likely pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>6448931</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>C</td>\n",
              "      <td>CG</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Duplication</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1</td>\n",
              "      <td>6451602</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>G</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>splice_acceptor_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5325</th>\n",
              "      <td>MT</td>\n",
              "      <td>15927</td>\n",
              "      <td>MT-TT</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5326</th>\n",
              "      <td>MT</td>\n",
              "      <td>15975</td>\n",
              "      <td>MT-TP</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5327</th>\n",
              "      <td>MT</td>\n",
              "      <td>15992</td>\n",
              "      <td>MT-TP</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5328</th>\n",
              "      <td>MT</td>\n",
              "      <td>15997</td>\n",
              "      <td>MT-TP</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5329</th>\n",
              "      <td>MT</td>\n",
              "      <td>16023</td>\n",
              "      <td>MT-TP</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5330 rows Ã— 10 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "     chr  position  ... biallelic      pathogenicity\n",
              "0      1   6425219  ...       yes         Pathogenic\n",
              "1      1   6425247  ...       yes  Likely pathogenic\n",
              "2      1   6445936  ...       yes  Likely pathogenic\n",
              "3      1   6448931  ...       yes         Pathogenic\n",
              "4      1   6451602  ...       yes         Pathogenic\n",
              "...   ..       ...  ...       ...                ...\n",
              "5325  MT     15927  ...       yes             Benign\n",
              "5326  MT     15975  ...       yes                NaN\n",
              "5327  MT     15992  ...       yes                NaN\n",
              "5328  MT     15997  ...       yes                NaN\n",
              "5329  MT     16023  ...       yes                NaN\n",
              "\n",
              "[5330 rows x 10 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 45
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "hLwAQc_R0Uem",
        "outputId": "b1ab75d9-0916-42b4-a626-cc046a237fd6"
      },
      "source": [
        "df_after_merge_annotate_summary[~df_after_merge_annotate_summary['pathogenicity'].isnull()]"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>chr</th>\n",
              "      <th>position</th>\n",
              "      <th>gene_names</th>\n",
              "      <th>ref</th>\n",
              "      <th>alt</th>\n",
              "      <th>alt_list</th>\n",
              "      <th>var_type</th>\n",
              "      <th>mol_consequence</th>\n",
              "      <th>biallelic</th>\n",
              "      <th>pathogenicity</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>6425219</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "      <td>yes</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>6425247</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>nonsense</td>\n",
              "      <td>yes</td>\n",
              "      <td>Likely pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>6445936</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>splice_donor_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>Likely pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>6448931</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>C</td>\n",
              "      <td>CG</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Duplication</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1</td>\n",
              "      <td>6451602</td>\n",
              "      <td>ESPN</td>\n",
              "      <td>G</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>splice_acceptor_variant</td>\n",
              "      <td>yes</td>\n",
              "      <td>Pathogenic</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5319</th>\n",
              "      <td>MT</td>\n",
              "      <td>14693</td>\n",
              "      <td>MT-TE</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5321</th>\n",
              "      <td>MT</td>\n",
              "      <td>15902</td>\n",
              "      <td>MT-TT</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5322</th>\n",
              "      <td>MT</td>\n",
              "      <td>15908</td>\n",
              "      <td>MT-TT</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5324</th>\n",
              "      <td>MT</td>\n",
              "      <td>15926</td>\n",
              "      <td>MT-TT</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5325</th>\n",
              "      <td>MT</td>\n",
              "      <td>15927</td>\n",
              "      <td>MT-TT</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>NaN</td>\n",
              "      <td>yes</td>\n",
              "      <td>Benign</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>4231 rows Ã— 10 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "     chr  position  ... biallelic      pathogenicity\n",
              "0      1   6425219  ...       yes         Pathogenic\n",
              "1      1   6425247  ...       yes  Likely pathogenic\n",
              "2      1   6445936  ...       yes  Likely pathogenic\n",
              "3      1   6448931  ...       yes         Pathogenic\n",
              "4      1   6451602  ...       yes         Pathogenic\n",
              "...   ..       ...  ...       ...                ...\n",
              "5319  MT     14693  ...       yes             Benign\n",
              "5321  MT     15902  ...       yes             Benign\n",
              "5322  MT     15908  ...       yes             Benign\n",
              "5324  MT     15926  ...       yes             Benign\n",
              "5325  MT     15927  ...       yes             Benign\n",
              "\n",
              "[4231 rows x 10 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 46
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "-rcQq2mT3Uu8"
      },
      "source": [
        "df_after_merge_annotate_summary.to_csv('/content/drive/MyDrive/GRAR/Paper_Surdez/Variants_after_merge_annotated.txt', index = None, header = True, sep = '\\t')"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "qJmkD7u2QFiO"
      },
      "source": [
        "### Detecting position on genes....\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "XVT0ntlw6i5-",
        "outputId": "6c65c5a9-8127-4fea-d5be-d4150b1e5b79"
      },
      "source": [
        "df_lost_pos = pd.read_table('/content/drive/MyDrive/GRAR/Paper_Surdez/regioes_corrigido.txt', usecols=['chr','start', 'stop'])\n",
        "#df_lost_pos['end'] = df_lost_pos['position'].astype(int) +1\n",
        "df_lost_pos.columns = ['Chromosome', 'Start', 'End']\n",
        "df_lost_pos['Chromosome'] = df_lost_pos['Chromosome'].str.replace('chr','')\n",
        "df_lost_pos = df_lost_pos.sort_values(by=['Chromosome', 'Start'])\n",
        "df_lost_pos = df_lost_pos.drop_duplicates()\n",
        "df_lost_pos"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "      <th>End</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>2009</th>\n",
              "      <td>1</td>\n",
              "      <td>789414</td>\n",
              "      <td>224014574</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2018</th>\n",
              "      <td>1</td>\n",
              "      <td>789416</td>\n",
              "      <td>224014443</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2027</th>\n",
              "      <td>1</td>\n",
              "      <td>789426</td>\n",
              "      <td>224014501</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2036</th>\n",
              "      <td>1</td>\n",
              "      <td>789430</td>\n",
              "      <td>224014567</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2045</th>\n",
              "      <td>1</td>\n",
              "      <td>789437</td>\n",
              "      <td>224014519</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2008</th>\n",
              "      <td>X</td>\n",
              "      <td>129526048</td>\n",
              "      <td>130517124</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>113990</th>\n",
              "      <td>X</td>\n",
              "      <td>130148900</td>\n",
              "      <td>130148900</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>113991</th>\n",
              "      <td>X</td>\n",
              "      <td>130148915</td>\n",
              "      <td>130148922</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>113995</th>\n",
              "      <td>X</td>\n",
              "      <td>130148915</td>\n",
              "      <td>130148916</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>113996</th>\n",
              "      <td>X</td>\n",
              "      <td>130150977</td>\n",
              "      <td>130150977</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5082 rows Ã— 3 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "       Chromosome      Start        End\n",
              "2009            1     789414  224014574\n",
              "2018            1     789416  224014443\n",
              "2027            1     789426  224014501\n",
              "2036            1     789430  224014567\n",
              "2045            1     789437  224014519\n",
              "...           ...        ...        ...\n",
              "2008            X  129526048  130517124\n",
              "113990          X  130148900  130148900\n",
              "113991          X  130148915  130148922\n",
              "113995          X  130148915  130148916\n",
              "113996          X  130150977  130150977\n",
              "\n",
              "[5082 rows x 3 columns]"
            ]
          },
          "metadata": {},
          "execution_count": 48
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 419
        },
        "id": "YXEW87-HoAp4",
        "outputId": "8d9e4588-72e1-4de8-a7b3-18af44dd1651"
      },
      "source": [
        "transcritos_pr = pr.PyRanges(df_transcritos_merged)\n",
        "df_lost_pos_pr = pr.PyRanges(df_lost_pos)\n",
        "\n",
        "transcritos_pr"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Gene</th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "      <th>End</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>ESPN</td>\n",
              "      <td>1</td>\n",
              "      <td>6424955</td>\n",
              "      <td>6460146</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GJB3</td>\n",
              "      <td>1</td>\n",
              "      <td>34784762</td>\n",
              "      <td>34785575</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>KCNQ4</td>\n",
              "      <td>1</td>\n",
              "      <td>40784093</td>\n",
              "      <td>40838523</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>BSND</td>\n",
              "      <td>1</td>\n",
              "      <td>54999186</td>\n",
              "      <td>55008628</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>ROR1</td>\n",
              "      <td>1</td>\n",
              "      <td>63774417</td>\n",
              "      <td>64178855</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>121</th>\n",
              "      <td>POU3F4</td>\n",
              "      <td>X</td>\n",
              "      <td>83508324</td>\n",
              "      <td>83509410</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>122</th>\n",
              "      <td>GPRASP2</td>\n",
              "      <td>X</td>\n",
              "      <td>102714869</td>\n",
              "      <td>102717386</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>123</th>\n",
              "      <td>PRPS1</td>\n",
              "      <td>X</td>\n",
              "      <td>107628628</td>\n",
              "      <td>107650032</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>124</th>\n",
              "      <td>COL4A6</td>\n",
              "      <td>X</td>\n",
              "      <td>108156999</td>\n",
              "      <td>108439371</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>125</th>\n",
              "      <td>AIFM1</td>\n",
              "      <td>X</td>\n",
              "      <td>130129361</td>\n",
              "      <td>130165841</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>126 rows Ã— 4 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "+------------+--------------+-----------+-----------+\n",
              "| Gene       | Chromosome   | Start     | End       |\n",
              "| (object)   | (category)   | (int32)   | (int32)   |\n",
              "|------------+--------------+-----------+-----------|\n",
              "| ESPN       | 1            | 6424955   | 6460146   |\n",
              "| GJB3       | 1            | 34784762  | 34785575  |\n",
              "| KCNQ4      | 1            | 40784093  | 40838523  |\n",
              "| BSND       | 1            | 54999186  | 55008628  |\n",
              "| ...        | ...          | ...       | ...       |\n",
              "| GPRASP2    | X            | 102714869 | 102717386 |\n",
              "| PRPS1      | X            | 107628628 | 107650032 |\n",
              "| COL4A6     | X            | 108156999 | 108439371 |\n",
              "| AIFM1      | X            | 130129361 | 130165841 |\n",
              "+------------+--------------+-----------+-----------+\n",
              "Unstranded PyRanges object has 126 rows and 4 columns from 23 chromosomes.\n",
              "For printing, the PyRanges was sorted on Chromosome."
            ]
          },
          "metadata": {},
          "execution_count": 49
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "G8igraOMoW3R"
      },
      "source": [
        "annotate_transcript = df_lost_pos_pr.join(transcritos_pr).drop(like=\"_b\")\n",
        "df_lost_pos_annot = annotate_transcript.df"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "VdV03BCG7BTk"
      },
      "source": [
        "#df_lost_pos_annot.groupby(['Chromosome','Start','End',])['Gene'].transform(lambda x: ','.join(x))\n",
        "#df_lost_pos_annot.groupby(['Chromosome','Start','End',])['Gene'].transform(lambda x: ','.join(x)).reset_index()\n",
        "df_lost_pos_annot.groupby(['Chromosome','Start','End',], as_index = False).agg({'Gene': ', '.join})\n",
        "#la.to_csv('/content/drive/MyDrive/GRAR/Paper_Surdez/All_positions_annottated.txt', index = None, header = True, sep = '\\t')"
      ],
      "execution_count": null,
      "outputs": []
    }
  ]
}